app/response_svc.py

```diff
@@ -85,25 +85,30 @@ async def refresh_blue_agents_abilities(self):
             if a not in self.abilities:
                 self.abilities.append(a)
 
-    async def run_abilities_on_agent(self, blue_agent, original_pid):
+    async def run_abilities_on_agent(self, blue_agent, red_agent, original_pid):
         facts = [Fact(trait='host.process.id', value=original_pid)]
         links = []
+        relationships = []
         for ability_id in self.abilities:
-            ability_facts, ability_links = await self.run_ability_on_agent(blue_agent, ability_id, facts, original_pid)
+            ability_facts, ability_links, ability_relationships = await self.run_ability_on_agent(blue_agent, red_agent, ability_id, facts, original_pid, relationships)
             links.extend(ability_links)
             facts.extend(ability_facts)
+            relationships.extend(ability_relationships)
```

Comment:
doesn't look like this relationship list is being used

Reply:
the `relationships` list is passed to the run_ability_on_agent at L93, the relationships are added after the ability's links are run, and these relationships are needed prior to running the next ability to do the filtering. We're unable to do the filtering between link creation and agent tasking because we use the agent's `task` function that creates the links and tasks the agent with them.